File: SpriteRectangle.java
import java.awt.*;
import java.awt.geom.AffineTransform;
import java.awt.geom.NoninvertibleTransformException;
import java.awt.geom.Point2D;
import java.awt.geom.RoundRectangle2D;

public class SpriteRectangle extends Sprite {
    private RoundRectangle2D shape = null;

    public SpriteRectangle(int width, int height, Sprite parent) {
        super(parent);
        this.shape = new RoundRectangle2D.Double(0, 0, width, height, 15, 15);
    }

    @Override
    public boolean containsPoint(Point2D p) {
        AffineTransform fullTransform = this.getFullTransform();
        AffineTransform inverseTransform = null;

        try {
            inverseTransform = fullTransform.createInverse();
        } catch (NoninvertibleTransformException e) {
            e.printStackTrace();
        }

        Point2D newPoint = inverseTransform.transform(p, null);
        return shape.contains(newPoint);
    }

    @Override
    protected void scaleShapeY(double scale) {
        //Should never reach here
    }

    @Override
    protected void startScale() {
        //Should never reach here
    }

    @Override
    protected void drawSprite(Graphics2D g) {
        g.setColor(this.fillColor);
        g.fill(shape);
        g.setColor(Color.black);
        g.draw(shape);
    }
}

File: SpriteRectangle.kt
import java.awt.*
import java.awt.geom.AffineTransform
import java.awt.geom.NoninvertibleTransformException
import java.awt.geom.Point2D
import java.awt.geom.RoundRectangle2D

class SpriteRectangle(width: Int, height: Int, parent: Sprite?) : Sprite(parent) {
    private var shape: RoundRectangle2D? = null

    init {
        this.shape = RoundRectangle2D.Double(0.0, 0.0, width.toDouble(), height.toDouble(), 15.0, 15.0)
    }

    override fun containsPoint(p: Point2D): Boolean {
        val fullTransform = this.fullTransform
        var inverseTransform: AffineTransform? = null

        try {
            inverseTransform = fullTransform.createInverse()
        } catch (e: NoninvertibleTransformException) {
            e.printStackTrace()
        }

        val newPoint = inverseTransform!!.transform(p, null)
        return shape!!.contains(newPoint)
    }

    override fun scaleShapeY(scale: Double) {
        //Should never reach here
    }

    override fun startScale() {
        //Should never reach here
    }

    override fun drawSprite(g: Graphics2D) {
        g.color = this.fillColor
        g.fill(shape)
        g.color = Color.black
        g.draw(shape)
    }
}
